azurelinuxagent/common/cgroupapi.py

```diff
@@ -275,6 +275,12 @@ def start_extension_command(self, extension_name, command, cmd_name, timeout, sh
 
             # We start systemd-run with shell == True so process.pid is the shell's pid, not the pid for systemd-run
             self._systemd_run_commands.append(process.pid)
+            # Extension run is child of the systemd-run and grandchild of the shell process.
+            ext_run_pid = osutil.get_extension_process_id(process.pid)
```

Comment:
we have a race condition here: the extension may not be started yet when Popen returns